internal/metrics/metrics.go

```diff
@@ -0,0 +1,174 @@
+package metrics
+
+import (
+	"strconv"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"sigs.k8s.io/controller-runtime/pkg/metrics"
+
+	lokiv1beta1 "github.com/ViaQ/loki-operator/api/v1beta1"
+)
+
+const (
+	CompactorEnabled  string = "compactor_enabled"
+	CompactorDisabled string = "compactor_disabled"
+	TenantLimits      string = "tenant_limits"
+	GlobalLimits      string = "global_limits"
+)
+
+var (
+	sizeTracker = prometheus.NewGaugeVec(
```

Comment:
nit. `Tracker` is an unusual suffix for metrics. I suggest to replace `Tracker` with `Metric`

Reply:
The scope of this variable is wide so I recommend `maxGlobalStreamsMetric` for a name